modules/fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/fs2/AttributedData.scala

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2020 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.enrich.fs2
+
+/** Data intended to be sent to a sink, plus key-value pairs to be added to the message as attributes */
+final case class AttributedData[A](data: A, attributes: Map[String, String])
```

Comment:
A very high-level note: `AttributedData` feels very related to `Payload`. I.e. `Payload` can be not just data + checkpoint, but data + checkpoint + metadata. Obviously not proposing to combine them now, but just saying this in the light of "FS2 data processing framework" idea - this primitive would be super common.